src/mailroom.py

```diff
@@ -0,0 +1,339 @@
+               'valid_responses': state_2_valid_responses,
+               }
+    state_3 = {'state': 3,
+               'comment': 'This is Print a Report of donor information.',
+               'prompt_message': '',
+               'user_response': '',
+               'valid_responses': state_3_valid_responses,
+               }
+    state_4 = {'state': 4,
+               'comment': 'Enter New Donation Amount',
+               'prompt_message': 'Enter New Donation Amount or' +
+               ' 0 to return to previous menu: ',
+               'donation_amount': 0.00,
+               'valid_responses': state_4_valid_responses,
+               }
+    state_5 = {'state': 5,
+               'comment': 'Ask to print a Thank You Letter',
+               'prompt_message': 'Print a thank you letter? (y or n): ',
+               'valid_responses': state_5_valid_responses,
+               }
+    state_6 = {'state': 6,
+               'comment': 'Ask to validate creating a user',
+               'prompt_message': ' (y or n): ',
+               'valid_responses': state_6_valid_responses,
+               'new_donor_name': ''
+               }
+    state_7 = {'state': 7,
+               'comment': 'Ask to validate donation amount',
+               'prompt_message': ' (y or n): ',
+               'valid_responses': state_7_valid_responses,
+               'new_donor_name': ''
+               }
+
+    state_dict = {
+        'state_1': state_1,
+        'state_2': state_2,
+        'state_3': state_3,
+        'state_4': state_4,
+        'state_5': state_5,
+        'state_6': state_6,
+        'state_7': state_7,
+    }
+
+    return state_dict
+
+
+def state_1_valid_responses(a):
+    global CURRENT_STATE
+    global STATE_DICT
+    if a == '0':
+        os.system('clear')
+        print('Good By')
+        sys.exit()
+    elif a == '1':
+        create_report()
+        CURRENT_STATE = STATE_DICT['state_1']
+        return True
+    elif a == '2':
+        CURRENT_STATE = STATE_DICT['state_2']
+        os.system('clear')
+        return True
+    else:
+        print('That was not a valid input')
+        return False
+
+
+def state_2_valid_responses(a):
+    ''' TODO: add check for new donor name greater then 20 characters '''
+    global CURRENT_STATE
+    global CURRENT_DONOR
+    if a == 'l' or a == 'list':
+        CURRENT_STATE = STATE_DICT['state_2']
+        list_of_donors(DONORS)
+    elif a == '0':
+        CURRENT_STATE = STATE_DICT['state_1']
+        os.system('clear')
+    else:
+        try:
+            CURRENT_DONOR = DONORS[a]
+            user_report()
+            CURRENT_STATE = STATE_DICT['state_4']
+        except KeyError:
+            STATE_DICT['state_2']['user_response'] = a
+            print('{} is not a current donor.'.format(a))
+            print('Create a new donor {}'.format(a), end='')
+            STATE_DICT['state_6']['new_donor_name'] = a
+            CURRENT_STATE = STATE_DICT['state_6']
+            return
```

Comment:
As things stand now, this function always returns None.  That means it does not operate the same way that the `state_1_valid_responses` function operates (that one returns True or False).  If the functions do not operate the same way, then they cannot be used in the same place as substitutes for each-other.  This makes it harder to build a generalized run loop that uses both.